### ISSUE TITLE
Getting an E_INVALIDARG when saving a weekly trigger

### ISSUE THREAD

**Issue by dahall** [dahall/TaskScheduler#95]
First up - I love this library, takes a lot of the pain out of programatically creating scheduled tasks.
 
I'm having a problem saving a task with a weekly trigger - when I call task.Definition.Triggers.Add(myTrigger), I'm getting the following exception:
 
 
System.ArgumentException was unhandled by user code
  Message=The parameter is incorrect. (Exception from HRESULT: 0x80070057 (E_INVALIDARG))
  Source=""
  StackTrace:
       at Microsoft.Win32.TaskScheduler.Trigger.Bind(ITaskDefinition iTaskDef)
       at Microsoft.Win32.TaskScheduler.TriggerCollection.Add(Trigger unboundTrigger)
       at Domain.Manager.HostedService.SaveTask(Task task) in C:\Current Projects\DataManager\Code\branches\UI.Client\Domain\Manager\HostedService.TaskService.cs:line 71
       at SyncInvokeSaveTask(Object , Object[] , Object[] )
       at System.ServiceModel.Dispatcher.SyncMethodInvoker.Invoke(Object instance, Object[] inputs, Object[]& outputs)
       at System.ServiceModel.Dispatcher.DispatchOperationRuntime.InvokeBegin(MessageRpc& rpc)
  InnerException: 
 
The error is not occurring when I try to save a DailyTrigger/MonthlyTrigger/MonthlyDOWTrigger; it doesn't seem to make much difference what values I put in for DaysOfWeek or WeeksInterval.

Originally posted: 2011-04-11T18:43:15

**Comment by dahall** [comment]
P.S. I am using v1.6.4097.31183 - nice work getting NuGet support in there, by the way - I came in this morning to find my code automagically updated to the new version.

Originally posted: 2011-04-11T18:54:46

**Comment by dahall** [comment]
Please post the code you are using to create the WeeklyTrigger. I know this code works:

td.Triggers.Add(new WeeklyTrigger(DaysOfTheWeek.Sunday, 3));

Originally posted: 2011-04-11T20:32:06

**Comment by dahall** [comment]
I've found the problem - I'd created a WCF-compatible library that maps to each of your classes (so adding DataContract/DataMember/EnumMember attributes to the classes/enums), and missed out a [DataMember] attribute on the WeeksInterval field, so it wasn't getting serialised and received at the server end when I mapped back to your classes.

It might be worth considering adding WCF support for a future release? There are quite a few hoops that I have to jump through with AutoMapper to fully create Task instances from the WCF messages (especially with the non-default constructors on some of the triggers).

Originally posted: 2011-04-11T22:48:07
